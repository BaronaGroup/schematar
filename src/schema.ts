import ObjectId from './object-id'
import Complex from './complex'

export type DefaultContext = 'mongoose' | 'typescript' | 'jsonschema'

export interface Schema {
    fields: SchemaFields
}

type SimpleType = Symbol | Complex | typeof String | typeof Number | typeof Boolean | typeof Date | typeof Object
type IndexOptions = false | true | 'unique' | 'unique-sparse'

export type PlainType = SimpleType | [FieldInfo | SimpleType]

export type Field = FieldInfo | PlainType

export interface FieldInfo {
    type: PlainType
    index?: IndexOptions
    format?: any
    optional?: boolean
    presentIn?: Array<string>
    optionalIn?: Array<string>
    enum?: string[]
    mongooseDefault?: any
    mongooseRef?: string
}

export interface SchemaFields {
    [name: string]: Field
}
