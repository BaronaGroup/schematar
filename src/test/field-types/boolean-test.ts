import {Schema} from '../../schema'
import {testJSONSchema, testMongooseField, testTypescriptInterface} from '../test-utils'

describe('boolean-test', function() {
  describe('basic type', function() {
    const schema: Schema = {fields: {
      field: Boolean
    }}

    it('json-schema', testJSONSchema(schema, data => {
      expect(data).toMatchSnapshot()
    }))

    it('mongoose', testMongooseField(schema, 'field', field => {
      expect(field).toEqual({type: Boolean})
    }))

    it('typescript', testTypescriptInterface(schema, tsInterface => {
      expect(tsInterface).toMatchSnapshot()
    }))
  })
})
