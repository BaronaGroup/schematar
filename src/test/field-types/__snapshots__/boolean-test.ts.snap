// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`boolean-test basic type json-schema 1`] = `
"{
  \\"type\\": \\"object\\",
  \\"$schema\\": \\"http://json-schema.org/draft-07/schema#\\",
  \\"properties\\": {
    \\"field\\": {
      \\"type\\": \\"boolean\\"
    }
  },
  \\"required\\": [
    \\"field\\"
  ]
}"
`;

exports[`boolean-test basic type typescript 1`] = `
"// Generated file, do not edit!

// tslint:disable
/* eslint-disable */
// @ts-ignore -- ignore possibly unused type parameters
export interface TestInterfaceBase<IDType, DateType>
{
  field: boolean
}"
`;
